src/redux/reducers/authReducer.js

```diff
@@ -28,6 +29,13 @@ export default (state = initialState, action) => {
         user: payload.user,
         isAuthenticated: true,
       };
+    case SOCIAL_LOGIN_SUCCESS:
+      return {
+        ...state,
+        loading: payload.loading,
+        user: payload,
+        isAuthenticated: true,
+      };
```

Comment:
Is there no way to combine `AUTH_SUCCESS` and `SOCIAL_LOGIN_SUCCESS` cases since they return the same state to avoid the repetition?

Reply:
I'll look at it again

Reply:
@mekzy-o have you fixed 👆🏽?

Reply:
I noticed our actions were not structured the same way partly due to how we were getting the user token. so I didn't use it again

Reply:
I don't think I get how that applies to combining two switch cases that return the same thing?

Reply:
I mean is the return same thing but not using the same structure:
```case AUTH_SUCCESS:
      return {
        ...state,
        loading: payload.loading,
        user: payload.user,
        isAuthenticated: true,
      };
    case SOCIAL_LOGIN_SUCCESS:
      return {
        ...state,
        loading: payload.loading,
        user: payload,
        isAuthenticated: true,
      };```
from the above code...you'd see I'm passing just `payload` to my reducer, while he is passing `payload.user`. That's the little difference

Reply:
Ok I see that now, thanks for pointing it out. 

But you should be able to conditionally return the appropriate value for `user`:
```
return {
        ...state,
        loading: payload.loading,
        user: payload.user ? payload.user : payload,
        isAuthenticated: true,
      };
```
You need to confirm that's valid though

Reply:
I've tried this, the conditional is a valid one but it doesn't work for me. The login functionalities of both social authentical and the normal login stopped working when I integrated it. The state was on a constant *loading: true*. Still looking at it to know if there is a work around